modules/Bio/EnsEMBL/Hive/Accumulator.pm

```diff
@@ -115,6 +115,8 @@ sub dataflow {
             my $key_signature = $accu_address;
             $key_signature=~s/(\w+)/$emitting_job->_param_possibly_overridden($1,$output_id)/eg;
```

Comment:
I was thinking about an alternative solution. First here:
```suggestion
            $key_signature=~s/(\w+)/$self->_check_empty_key($1, $emitting_job->_param_possibly_overridden($1,$output_id))/eg;
```
And something like this further down in the module:
```perl
sub _check_empty_key {
    my ($key, $value) = @_;
    if ((not defined $value) || ($value eq '')) {
        die "No value provided for $key. It is needed for the accu dataflow.";
    }
}
```

The function call is only activated when `\w+` is found, which should be for all hash keys/array indexes.
Something to check though: is the `die` going to break the `s///` substitution and "bubble" up ?

Reply:
Regexp is possible, but current solution it's more maintanable and readable, and we give more information for user in error - which exacly brackets brakes a job.